### PR TITLE
feat: allow configuration of cors domains and credentials

### PR DESCRIPTION
**Pull request by jhleao**
## Problem
For applications that use cookie-based auth, CORS domain wildcards are not allowed. Need to specify explicit domains instead. Plus, `Access-Control-Allow-Credentials` must be set to `true`.

## Solution
This makes CORS domains optionally configurable via `CORS_ALLOW_ORIGINS`, same for `Access-Control-Allow-Credentials` via `CORS_ALLOW_CREDENTIALS`. If no configuration is specified, current behavior is kept.

### CODE REVIEW

**Comment by dbarrosop** [comment]
Thanks for the PR. However, we need to slightly change the implementation:

1. Instead or reading directly from env vars command flags (i.e. `cors-allow-origins`), defaulting to the current value `*`.
2. Pass the value of the resolved flags from `cmd` down to `SetupRouter`

Thanks again for the contribution.

**Comment by dbarrosop** [comment]
Thanks! Looks good to me but tests need to be fixed, I think you just need to pass the new parameters to `SetupRouter` in the tests that use it.

**Comment by jhleao** [comment]
@dbarrosop I couldn't get these tests to pass locally... But I believe the problem to be on my dev environment. Can you trigger CI again?

File: controller/delete_broken_metadata_test.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func TestDeleteBrokenMetadata(t *testing.T) {
 
 			ctrl := controller.New("http://asd", "/v1", "asdasd", metadataStorage, contentStorage, nil, logger)
 
-			router, _ := ctrl.SetupRouter(nil, "/v1", ginLogger(logger))
+			router, _ := ctrl.SetupRouter(nil, "/v1", []string{}, false, ginLogger(logger))
```
Comment (dbarrosop) [inline controller/delete_broken_metadata_test.go:106]: I think you need to set this to `[]string{"*"}` which was the previous setting based on your changes.

**Comment by dbarrosop** [comment]
Nice! Thanks a lot for the contribution.

**Pull request merged**